src/common/http/http.cc

```diff
@@ -621,6 +622,71 @@ response_t HttpImpl::Get(
   }
 }
 
+response_t HttpImpl::SimpleGet(
+    absl::string_view uri,
+    const std::map<absl::string_view, absl::string_view> &headers,
+    absl::string_view body, boost::asio::io_context &ioc,
+    boost::asio::yield_context yield) const {
+  spdlog::trace("{}", __func__);
+  try {
+    int version = 11;
+
+    auto parsed_uri = http::Uri(uri);
+
+    tcp::resolver resolver(ioc);
+    beast::tcp_stream stream(ioc);
+
+    spdlog::info("{}: opening connection to {}:{}", __func__,
+                 parsed_uri.GetHost(), parsed_uri.GetPort());
+    const auto results = resolver.async_resolve(
+        parsed_uri.GetHost(), std::to_string(parsed_uri.GetPort()), yield);
+    beast::get_lowest_layer(stream).async_connect(results, yield);
+
+    // Set up an HTTP POST request message
+    beast::http::request<beast::http::string_body> req{
+        beast::http::verb::get, parsed_uri.GetPathQueryFragment(), version};
+    req.set(beast::http::field::host, parsed_uri.GetHost());
+    for (auto header : headers) {
+      req.set(boost::beast::string_view(header.first.data()),
+              boost::beast::string_view(header.second.data()));
+    }
+    auto &req_body = req.body();
+    req_body.reserve(body.size());
+    req_body.append(body.begin(), body.end());
+    req.prepare_payload();
+    // Send the HTTP request to the remote host
+    beast::http::async_write(stream, req, yield);
+
+    // Read response
+    beast::flat_buffer buffer;
+    response_t res(new beast::http::response<beast::http::string_body>);
+    beast::http::async_read(stream, buffer, *res, yield);
+
+    // Gracefully close the socket.
+    // Receive an error code instead of throwing an exception if this fails, so
+    // we can ignore some expected not_connected errors.
+    boost::system::error_code ec;
+
+    if (ec) {
+      // not_connected happens sometimes so don't bother reporting it.
+      // stream_truncated also happens sometime and we choose to ignore the
+      // stream_truncated error, as recommended by the github thread:
+      // https://github.com/boostorg/beast/issues/824
+      if (ec != beast::errc::not_connected &&
+          ec != boost::asio::ssl::error::stream_truncated) {
+        spdlog::error("{}: HTTP error encountered: {}", __func__, ec.message());
+        return response_t();
+      }
+    }
+
+    return res;
+    // If we get here then the connection is closed gracefully
+  } catch (std::exception const &e) {
+    spdlog::error("{}: unexpected exception: {}", __func__, e.what());
+    return response_t();
```

Comment:
this will return a default response_t to the caller. and the exception is handled within the simpleGet. how do we expect the caller to check whether return value is legit or not? absl::StatusOr seems a good fit for this.